rpcs3/Input/ds4_pad_handler.cpp

```diff
@@ -906,6 +919,25 @@ void ds4_pad_handler::apply_pad_data(const std::shared_ptr<PadDevice>& device, c
 		newBlinkData = true;
 	}
 
+	// Use LEDs to indicate battery level
+	if (false) // TODO: Implement a setting to enable this. The following code works and correctly indicates the current battery level by changing the LED color.
+	{
+		static const std::array<u32, 11> battery_level_clr = {0xff00, 0xff33, 0xff66, 0xff99, 0xffcc, 0xffff, 0xccff, 0x99ff, 0x66ff, 0x33ff, 0x00ff}; // TODO: Move this elsewhere. Maybe even allow custom gradients?
+		static u32 last_battery_level = 11;
```

Comment:
this won't work for multiple controllers I guess

Reply:
Are you sure? `apply_pad_data` receives a specific device to work with, hence this should execute for each controller separately. Will test though.

Reply:
After testing with multiple controllers, this seems to work but needs further testing. It gets overridden with default or saved LED settings on pad initialization. This is expected and is to be fixed once the function is properly implemented as an optional setting.

Reply:
https://stackoverflow.com/a/6223371
You basically use the same variable for all controllers. Which means you will always get LED updates on two controllers if both (or more) controllers have different values. Even if they themselves didn't change battery levels.

Reply:
Thank you for pointing this out, adjusted it to instead use a field in a DS4Device struct.